docs/feature_autocorrect.md

```diff
@@ -0,0 +1,295 @@
+# Autocorrect
+
+There are a lot of words that are prone to being typed incorrectly, due to habit, sequence or just user error.  This feature leverages your firmware to automatically correct these errors, to help reduce typos.
+
+## How does it work? :id=how-does-it-work
+
+The feature maintains a small buffer of recent key presses. On each key press, it checks whether the buffer ends in a recognized typo, and if so, automatically sends keystrokes to correct it.
+
+The tricky part is how to efficiently check the buffer for typos. We don’t want to spend too much memory or time on storing or searching the typos. A good solution is to represent the typos with a trie data structure. A trie is a tree data structure where each node is a letter, and words are formed by following a path to one of the leaves.
+
+![An example trie](https://getreuer.info/posts/keyboards/autocorrection/trie.svg)
+
+Since we search whether the buffer ends in a typo, we store the trie writing in reverse. The trie is queried starting from the last letter, then second to last letter, and so on, until either a letter doesn’t match or we reach a leaf, meaning a typo was found.
+
+## How do I enable Autocorrection :id=how-do-i-enable-autocorrection
+
+In your `rules.mk`, add this: 
+
+```make
+AUTOCORRECT_ENABLE = yes
+```
+
+Additionally, you will need a library for autocorrection.  A small sample library is included by default, so that you can get up and running right away, but you can provide a customized library.   
+
+By default, autocorrect is disabled.  To enable it, you need to use the `AUTOCORRECT_TOGGLE` keycode to enable it. The status is stored in persistent memory, so you shouldn't need to enabled it again.
```

Comment:
Well, a simpler answer would be to set the default. 

I don't know if you saw, but the oneshot setting got tweaked again.  Because like you said, negative names get confusing, and the toggle wasn't working correct for oneshots.   But set the default bit to enable oneshots by default. Can do the same for autocorrect. 

Also, setting the status in eeconfig_init_user would be a better option, so it's not as brute force.

Reply:
That's fantastic! I'm glad that defaulting config bits to 1 is an option.

PR https://github.com/qmk/qmk_firmware/pull/16617 seemed to fix oneshots on my keyboard, but sounds like I overlooked something. Thanks for cleaning it up 😁

Reply:
> Also, setting the status in eeconfig_init_user would be a better option, so it's not as brute force.

I'm thinking if someone is trying but struggling to get autocorrect to work, having a "brute force recipe" to enable autocorrect would help. Enabling autocorrect is definitely a step easier now with https://github.com/qmk/qmk_firmware/pull/15699/commits/024a167fc665ca1d3a33d6eecc9268a33618d663, thanks for making that! 

Now, a user could (I think) ensure autocorrect on by clearing EEPROM through something like the Command magic key. What do you think about adding an "`AUTOCORRECT_ON`" keycode? I don't feel super strongly, but it seems worthwhile to make this easier yet.

Reply:
> That's fantastic! I'm glad that defaulting config bits to 1 is an option.
> 
> PR #16617 seemed to fix oneshots on my keyboard, but sounds like I overlooked something. Thanks for cleaning it up 😁

Toggle wasn't working correctly.  And instead of making it more convoluted and confusing to read ... 

> I'm thinking if someone is trying but struggling to get autocorrect to work, having a "brute force recipe" to enable autocorrect would help. Enabling autocorrect is definitely a step easier now with https://github.com/qmk/qmk_firmware/commit/024a167fc665ca1d3a33d6eecc9268a33618d663, thanks for making that!
> Now, a user could (I think) ensure autocorrect on by clearing EEPROM through something like the Command magic key. What do you think about adding an "AUTOCORRECT_ON" keycode? I don't feel super strongly, but it seems worthwhile to make this easier yet.

That shouldn't be a problem, and would fall more in line with a lot of the existing features, too.

Reply:
> Toggle wasn't working correctly. And instead of making it more convoluted and confusing to read ...

D'oh! Thanks, I see it now.

> That shouldn't be a problem, and would fall more in line with a lot of the existing features, too.

This is great. Thanks a bunch!